src/resource-rental/resource-rental.ts

```diff
@@ -74,7 +76,19 @@ export class ResourceRental {
           }
 
           this.logger.info("Waiting for payment process of agreement to finish", { agreementId: this.agreement.id });
-          await waitForCondition(() => this.paymentProcess.isFinished());
+          const abortSignal = createAbortSignalFromTimeout(signalOrTimeout);
+          await waitForCondition(() => this.paymentProcess.isFinished(), {
+            signalOrTimeout: abortSignal,
+          }).catch((error) => {
+            this.paymentProcess.stop();
+            if (error instanceof GolemTimeoutError) {
+              throw new GolemTimeoutError(
+                `The finalization of payment process has been aborted due to a timeout (${signalOrTimeout}ms)`,
```

Comment:
isn't `${signalOrTimeout}ms` going to end up as `[object AbortSignal]ms`?

Reply:
Actually, there is probably only one situation in which the type can be `[object AbortSignal]`, when the user provides `AbortSignal.timeout()` as a parameter. In all other cases, `GolemTimeoutError` thrown by `waitForCondition` will only occur if the user specifies `signalOrTimeout` as a number. But I guess because of this one exception we have to remove this information about the timeout value.